client/hub/client_test.go

```diff
@@ -135,21 +198,80 @@ func TestQueryWithTanzuHubClient(t *testing.T) {
 			mockResponses: []hubtesting.Operation{
 				{
 					Identifier: "QueryAllProjects",
-					Responder: func(ctx context.Context, op hubtesting.Operation) hub.Response {
+					// Change the value of the default variables
+					// to make sure the mock server returns the variables that were
+					// received in the query and NOT the ones that are registered here.
+					// Note that the variable keys must match the ones in the query, but
+					// the values can be different.
+					Variables: map[string]interface{}{
+						"name":  "notjohn",
+						"lines": 0,
+					},
+					Responder: func(_ context.Context, receivedReq hubtesting.Request) hub.Response {
 						return hub.Response{
-							Errors: gqlerror.List{{Message: fmt.Sprintf("operation %s failed with error %s", op.Identifier, "fake-error-message")}},
+							Errors: gqlerror.List{{Message: fmt.Sprintf("operation failed with error and received %v", receivedReq)}},
```

Comment:
nit(minor): I am not sure if updating the message as below will help to understand things a bit more or not. Ignore if you feel it does not make sense.

```suggestion
							Errors: gqlerror.List{{Message: fmt.Sprintf("operation failed with error when received request as: %v", receivedReq)}},
```